src/pipecat/pipeline/llm_switcher.py

```diff
@@ -0,0 +1,156 @@
+#
+# Copyright (c) 2025, Daily
+#
+# SPDX-License-Identifier: BSD 2-Clause License
+#
+
+"""LLM switcher for switching between different LLMs at runtime, with different switching strategies."""
+
+from typing import Any, Generic, List, Optional, Type, TypeVar
+
+from pipecat.pipeline.parallel_pipeline import ParallelPipeline
+from pipecat.processors.aggregators.llm_context import LLMContext
+from pipecat.processors.filters.function_filter import FunctionFilter
+from pipecat.processors.frame_processor import FrameDirection
+from pipecat.services.llm_service import LLMService
+
+
+class LLMSwitcherStrategy:
+    """Base class for LLM switching strategies."""
+
+    def __init__(self, llms: List[LLMService]):
+        """Initialize the LLM switcher strategy with a list of LLM services."""
+        self.llms = llms
+        self.active_llm: Optional[LLMService] = None
+
+    def is_active(self, llm: LLMService) -> bool:
+        """Determine if the given LLM is the currently active one.
+
+        This method should be overridden by subclasses to implement specific logic.
+
+        Args:
+            llm: The LLM service to check.
+
+        Returns:
+            True if the given LLM is the active one, False otherwise.
+        """
+        raise NotImplementedError("Subclasses must implement this method.")
+
+
+StrategyType = TypeVar("StrategyType", bound=LLMSwitcherStrategy)
+
+
+class LLMSwitcher(ParallelPipeline, Generic[StrategyType]):
+    """A pipeline that switches between different LLMs at runtime."""
+
+    def __init__(self, llms: List[LLMService], strategy_type: Type[StrategyType]):
+        """Initialize the LLM switcher with a list of LLM services and a switching strategy."""
+        strategy = strategy_type(llms)
+        super().__init__(*LLMSwitcher._make_pipeline_definitions(llms, strategy))
+        self.llms = llms
+        self.strategy = strategy
+
+    async def generate_summary(self, summary_prompt: str, context: LLMContext) -> Optional[str]:
```

Comment:
If we make this generic (i.e. not summary-specific) in the `LLMService`, match that here.